File: php/testcode/ldapi_041.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_ldapi_enum_attribute
function ldapi041(BenchmarkRequest $req): BenchmarkResponse {
    $conn = ldap_connect('ldap://localhost');
    $base = 'dc=example,dc=com';
    $choice = (int)$req->param('choice');
    $attr = ['uid', 'cn', 'mail'][$choice] ?? 'uid'; // vuln-code-snippet safe-line php_ldapi_enum_attribute
    $filter = "($attr=*)";
    $result = ldap_search($conn, $base, $filter);
    $entries = ldap_get_entries($conn, $result);
    return BenchmarkResponse::json(['count' => $entries['count']]);
}
// vuln-code-snippet end php_ldapi_enum_attribute

File: rust/testcode/ldapi_041.rs
//! CWE-90: Dead-code branch — constant-true condition always selects safe hardcoded filter.

fn ldap_search(base: &str, filter: &str) -> String {
    // In production: ldap3::LdapConn::new("ldap://...").search(base, Scope::Subtree, filter, vec!["cn", "mail"])
    format!("LDAP search in {} with filter {}", base, filter)
}

// vuln-code-snippet start testcodeLdapi041
pub fn handle(req: &super::shared::BenchmarkRequest) -> super::shared::BenchmarkResponse {
    let user = req.param("username");
    let filter = if 7 * 6 > 40 {
        "(objectClass=person)".to_string()
    } else {
        format!("(uid={})", user)
    };
    let result = ldap_search("dc=example,dc=com", &filter); // vuln-code-snippet target-line testcodeLdapi041
    super::shared::BenchmarkResponse::ok(&result)
}
// vuln-code-snippet end testcodeLdapi041
